# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382193239 25200
#      Sat Oct 19 07:33:59 2013 -0700
# Node ID 9bb5d3f63cdd9cc079813a06edd3416f601c9178
# Parent  f0e777cf348f385b93bb360d749867c70c94c1b1
NEWS: Announce change in definition of kurtosis function for Matlab compatibility.

* NEWS: Announce that kurtosis no longer calculates "excess kurtosis".

* scripts/statistics/base/kurtosis.m: Tweak docstring.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -256,16 +256,21 @@ Summary of important user-visible change
     In addition two new error functions erfi (imaginary error function) and
     dawson (scaled imaginary error function) have been added.
 
  ** The glpk function has been modified to reflect changes in the GLPK
     library.  The "round" and "itcnt" options have been removed.  The
     "relax" option has been replaced by the "rtest" option.  The numeric
     values of error codes and of some options have also changed.
 
+ ** The kurtosis function has changed definition to be compatible with 
+    Matlab.  It now returns the base kurtosis instead of the "excess kurtosis".
+    The old behavior can be had by changing scripts to normalize with -3.
+               "excess kurtosis" = kurtosis (x) - 3
+
  ** The default name of the Octave crash dump file is now 
     "octave-workspace" instead of "octave-core".
 
  ** A citation command has been added to display information on how to
     cite Octave and packages in publications.  The package system will
     look for and install CITATION files from packages.
 
  ** The java package from Octave Forge is now part of core Octave.  The
diff --git a/scripts/statistics/base/kurtosis.m b/scripts/statistics/base/kurtosis.m
--- a/scripts/statistics/base/kurtosis.m
+++ b/scripts/statistics/base/kurtosis.m
@@ -22,55 +22,55 @@
 ## @deftypefnx {Function File} {} kurtosis (@var{x}, @var{flag})
 ## @deftypefnx {Function File} {} kurtosis (@var{x}, @var{flag}, @var{dim})
 ## Compute the sample kurtosis of the elements of @var{x}:
 ## @tex
 ## $$
 ## \kappa_1 = {{{1\over N}\,
 ##          \sum_{i=1}^N (@var{x}_i - \bar{@var{x}})^4} \over \sigma^4},
 ## $$
-## where $N$ is the length of @var{x}, $\bar{@var{x}}$ its mean and $\sigma$
+## where $N$ is the length of @var{x}, $\bar{@var{x}}$ its mean, and $\sigma$
 ## its (uncorrected) standard deviation.
 ## @end tex
 ## @ifnottex
 ##
 ## @example
 ## @group
 ##      mean ((@var{x} - mean (@var{x})).^4)
-## k1 = ------------------------.
+## k1 = ------------------------
 ##             std (@var{x}).^4
 ## @end group
 ## @end example
 ##
 ## @end ifnottex
 ##
 ## @noindent
 ## The optional argument @var{flag} controls which normalization is used.
 ## If @var{flag} is equal to 1 (default value, used when @var{flag} is omitted
 ## or empty), return the sample kurtosis as defined above.  If @var{flag} is
-## equal to 0, return the "bias-corrected" kurtosis coefficient instead:
+## equal to 0, return the @w{"bias-corrected"} kurtosis coefficient instead:
 ## @tex
 ## $$
 ## \kappa_0 = 3 + {\scriptstyle N - 1 \over \scriptstyle (N - 2)(N - 3)} \,
-##     \left( (N + 1)\, \kappa_1 - 3 (N - 1) \right).
+##     \left( (N + 1)\, \kappa_1 - 3 (N - 1) \right)
 ## $$
 ## @end tex
 ## @ifnottex
 ##
 ## @example
 ## @group
 ##               N - 1
 ## k0 = 3 + -------------- * ((N + 1) * k1 - 3 * (N - 1))
 ##          (N - 2)(N - 3)
 ## @end group
 ## @end example
 ##
 ## @end ifnottex
 ## The bias-corrected kurtosis coefficient is obtained by replacing the sample
-## second and fourth central moments by their unbiased versions. It is an
+## second and fourth central moments by their unbiased versions.  It is an
 ## unbiased estimate of the population kurtosis for normal populations.
 ##
 ## If @var{x} is a matrix, or more generally a multi-dimensional array, return
 ## the kurtosis along the first non-singleton dimension.  If the optional
 ## @var{dim} argument is given, operate along this dimension.
 ##
 ## @seealso{var, skewness, moment}
 ## @end deftypefn
